extensions/ql-vscode/src/pure/interface-types.ts

```diff
@@ -44,7 +45,14 @@ export interface PreviousExecution {
   durationSeconds: number;
 }
 
-export interface Interpretation {
+export type SarifInterpretationData = {
+  t: 'SarifInterpretationData';
```

Comment:
Here, you are added a nested `t` field. This seems to complicate things somewhat. Would it make sense to keep only one `t` field at the top level, and specify `RawResultSet`, `SarifResultSet`, and (eventually) `GraphResultSet`?

I don't remember your other PR, so maybe there's a good reason to keep this.

Reply:
This is in preparation for
```ts
export type GraphInterpretationData = {
  t: 'GraphInterpretationData';
  dot: string[];
};
```
(from [the other PR](https://github.com/github/vscode-codeql/pull/705)) so we can do dispatch on the types (this is how to do tagged unions in Typescript, right?)